### PR TITLE
Resolves #3293: Posts Service Documentation

### LINKED ISSUE #3293: Write `posts` service documentation
Part of #3291, posts service was my service so I'd love to do this revamp

### PR DESCRIPTION
**Pull request by HyperTHD**
## Issue This PR Addresses
This PR closes #3293

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [x] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

This PR adds documentation for all routes for the posts service. Based on #1907, I decided not to include the feeds routes in the documentation. This can be added if needed however until documentation for the parser service is complete.

## Steps to test the PR

No testing required, simply leave feedback on whether I got everything here related to the posts service's routes

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [x] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by JiaHua-Zou — APPROVED** [review]
Looks good. The The json codeblock at the bottom of the page is red, but I check it on docusaurus and it's fine.

File: src/web/docusaurus/docs/api-services/posts.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+##### Successful response
+
+```
+Status: 200 OK
+```
+
+```json
+{
+  id: '28867e65b0',
```
Comment (AmasiaNalbandian) [inline src/web/docusaurus/docs/api-services/posts.md:95]: I'm not sure why this is in red highlight? Is that just my computer?

Comment (HyperTHD) [reply src/web/docusaurus/docs/api-services/posts.md:95]: I forgot to put quotes around the key name, it's red cause it's not valid in json, gonna fix that

**Review by tpmai22 — APPROVED** [review]
LGTM !

**Review by JiaHua-Zou — APPROVED** [review]

**Review by humphd — APPROVED** [review]

**Pull request merged**